Define FAR to nothing for the sake of krb5 1.3.  Declare the proper
preprocessor directives to force declaration of internal and deprecated
interfaces (the right thing to do is to port it, but this might work).

--- openssl-0.9.7a/ssl/kssl.h	2003-07-17 17:54:33.000000000 -0400
+++ openssl-0.9.7a/ssl/kssl.h	2003-07-17 17:54:35.000000000 -0400
@@ -81,8 +81,10 @@
 */
 #ifdef KRB5_HEIMDAL
 typedef unsigned char krb5_octet;
-#define FAR
 #endif
+#ifndef FAR
+#define FAR
+#endif
 
 /*	Uncomment this to debug kssl problems or
 **	to trace usage of the Kerberos session key
--- openssl-0.9.7a/ssl/kssl.c	2003-07-17 17:55:40.000000000 -0400
+++ openssl-0.9.7a/ssl/kssl.c	2003-07-17 18:00:35.000000000 -0400
@@ -66,6 +66,13 @@
 **          		Real RFC2712 KerberosWrapper replaces AP_REQ.
 */
 
+#ifndef OPENSSL_NO_KRB5
+
+#include <errno.h>
+/* Signal 1.3 headers to declare bits and pieces of the 1.2 API. */
+#define KRB5_PRIVATE 1
+#define KRB5_DEPRECATED 1
+
 #include <openssl/opensslconf.h>
 
 #define _XOPEN_SOURCE /* glibc2 needs this to declare strptime() */
@@ -77,8 +84,6 @@
 #include <openssl/objects.h>
 #include <openssl/krb5_asn.h>
 
-#ifndef OPENSSL_NO_KRB5
-
 /* 
  * When OpenSSL is built on Windows, we do not want to require that
  * the Kerberos DLLs be available in order for the OpenSSL DLLs to
